src/figuresClasses.ts

```diff
@@ -1,19 +1,79 @@
-export interface Figure {
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
 
-}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('All sides should be positive numbers');
+    }
 
-export class Circle {
+    if (Math.max(a, b, c) >= (a + b + c) - (Math.max(a, b, c))) {
+      throw new Error(`sides ${a}, ${b} and ${c}`);
+    }
+  }
 
+  getArea(): number {
+    const semiPerimeter: number = (this.a + this.b + this.c) / 2;
+    const underTheRoot: number = semiPerimeter * (semiPerimeter - this.a)
+      * (semiPerimeter - this.b) * (semiPerimeter - this.c);
+    const square: number = Math.sqrt(underTheRoot);
+
+    return Math.floor(square * 100) / 100;
+  }
 }
 
-export class Rectangle {
+export class Circle implements Figure {
+  shape: Shape = 'circle';
+
+  constructor(
+    public color: Color,
+    public radius: number,
+  ) {
+    if (radius <= 0) {
+      throw new Error('Radius should be a positive numbers');
+    }
+  }
 
+  getArea(): number {
+    const square: number = Math.PI * this.radius ** 2;
```

Comment:
avoid types here, because the result of your calc would be a number
```suggestion
    const square = Math.PI * this.radius ** 2;
```